### PR TITLE
Deprecate `Trainer.verbose_evaluate`

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Deprecates `Trainer.vebose_evaluate` in favor of `EvaluationLoop(verbose: bool = True)`.

Follow-up to #10928

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @justusschock @awaelchli @akihironitta @tchaton @borda @carmocca @ananthsub @ninginthecloud

### CODE REVIEW

**Review by justusschock — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -161,6 +161,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Deprecated `Trainer.run_stage` in favor of `Trainer.{fit,validate,test,predict}` ([#11000](https://github.com/PyTorchLightning/pytorch-lightning/pull/11000))
 
 
+- Deprecated `Trainer.verbose_evaluate` in favor of `EvaluationLoop(verbose=...)` ([#10931](https://github.com/PyTorchLightning/pytorch-lightning/pull/10931))
```
Comment (awaelchli) [inline CHANGELOG.md:164]: I would mention the verbose arg in Trainer.validate/test because that's more usable compared to instantiating the custom loop.

File: pytorch_lightning/trainer/trainer.py [hunk 2]
```diff
@@ -462,9 +462,6 @@ def __init__(
         )
         fit_loop = FitLoop(min_epochs=min_epochs, max_epochs=max_epochs)
         training_epoch_loop = TrainingEpochLoop(min_steps=min_steps, max_steps=max_steps)
-        training_batch_loop = TrainingBatchLoop()
-        training_validation_loop = EvaluationLoop()
-        training_epoch_loop.connect(batch_loop=training_batch_loop, val_loop=training_validation_loop)
```
Comment (awaelchli) [inline pytorch_lightning/trainer/trainer.py:467]: this is unrelated?

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (ananthsub) [reply CHANGELOG.md:164]: out of curiosity, why do we have verbose on validate and test but not on fit?

File: pytorch_lightning/trainer/trainer.py (hunk 2 above)
Comment (carmocca) [reply pytorch_lightning/trainer/trainer.py:467]: It's not, this was unsetting the `training_validation_loop` value that the `training_epoch_loop` sets. Setting these is not necessary.

Another option would have been to also add `verbose=False` here

File: CHANGELOG.md (hunk 1 above)
Comment (carmocca) [reply CHANGELOG.md:164]: Because fit does not return `something = trainer.fit(...)`

But I don't know why that's the case
